### PR TITLE
Put puts a metric on directly on the datapoint channel.

### PR DESCRIPTION
**Pull request by kylebrandt**
This is useful for capturing "events" that have a gauge value

### CODE REVIEW

**Comment by kylebrandt** [comment]
Example usage:

```
func (s *Schedule) CheckAlert(r RunHistory, a *conf.Alert) {
    ...
    collect.Put("check.duration", opentsdb.TagSet{"name": a.Name}, time.Now().Unix(), time.Since(start).Seconds())
    ...
}
```

**Comment by madelynnblue** [comment]
I think this function is unsafe because it could be called twice within the same second and thus produce OpenTSDB data errors if the values differ between the two calls.

**Comment by kylebrandt** [comment]
Have a different idea?

**Comment by madelynnblue** [comment]
Register a Set handler. This has the negative side effects that it can both miss events (if the sample rate isn't high enough) or report duplicate data (if it's sampled between events), but OpenTSDB isn't an event database, so we already have those problems when measuring events. I don't yet think this needs a new API call.

File: collect/collect.go [hunk 1]
```diff
@@ -194,6 +195,18 @@ func Add(metric string, ts opentsdb.TagSet, inc float64) error {
 	return nil
 }
 
+// Put is useful for capturing "events" that have a gauge value. Subsequent calls between the sending interval will overwrite previous calls
```
Comment (madelynnblue) [inline collect/collect.go:198]: Wrap this comment to 80 chars per line.

**Comment by madelynnblue** [comment]
LGTM

**Pull request merged**